src/process.c

```diff
@@ -17,7 +17,9 @@ extern "C"
 {
 #endif
 
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
```

Comment:
This concerns me.  In particular, I'm worried that we now may start using incompatible symbols *within* rcutils, if we forget to define this everywhere it is needed.  If we are switching to the GNU API, then I'd suggest we just go all the way and do `add_compile_definitions(-D_GNU_SOURCE)` unconditionally for all files in here.  We may also consider the same for other C libraries, though we can deal with them on a case-by-case basis.  Thoughts?

Reply:
Sorry, I totally misread this patch.  Above you are doing exactly what I expect; `add_definitions(-D _GNU_SOURCE)`.  With that being the case, I would say we can just remove this `#ifndef` block completely.  It will even be "more correct" without this patch on platforms other than Linux, as we'll now only set `_GNU_SOURCE` on Linux.